c++/src/capnp/compiler/capnp.c++

```diff
@@ -502,6 +502,23 @@ public:
       requestedFile.setFilename(sourceFiles[i].name);
       requestedFile.adoptImports(compiler->getFileImportTable(
           *sourceFiles[i].module, Orphanage::getForMessageContaining(requestedFile)));
+      // Add FileSourceInfo
+      auto fileSourceInfo = requestedFile.initFileSourceInfo();
+      const auto resolutions = sourceFiles[i].module->getResolutions();
+      auto identifiers = fileSourceInfo.initIdentifiers(resolutions.size());
+      for (size_t j = 0; j < resolutions.size(); j++) {
+        auto identifier = identifiers[j];
+        identifier.setStartByte(resolutions[j].startByte);
+        identifier.setEndByte(resolutions[j].endByte);
+        if (resolutions[j].target.is<Resolution::Type>()) {
```

Comment:
Can you use a KJ_SWITCH_ONEOF here?

```
KJ_SWITCH_ONEOF(resolutions[j].target) {
  KJ_CASE_ONEOF(type, Resolution::Type) {
    identifier.setTypeId(type.typeId);
  }
  KJ_CASE_ONEOF(member, Resolution::Member) {
    ...
  }
}
```

Reply:
Thank you for your advice. I modified it as you mentioned.